### ISSUE TITLE
Questions about the results

### ISSUE THREAD

**Issue by NAZABANMAAFK** [MediaBrain-SJTU/RegAD#34]
Dear author, thank you very much for your work! I am surprised and very interested in your ideas.After carefully reading your paper and reproducing the code, I am puzzled by the following two points：
Firstly, why do we perform inference before every training time?

<img width="710" alt="c19d34d0af2d5333abc0437eeb391c0" src="https://user-images.githubusercontent.com/98138999/230551962-9a90d3ec-9b8b-4856-87f5-9414ff84eec0.png">
In the test function, we passed through the feature extraction network, STN, siamese, and prediction network, and returned four values for calculating AUROC，after that, we started to train, and the training process is consistent with that in the test function.I noticed that AUROC did not have an impact on subsequent training during this process. What is the purpose of doing so? Can I directly select a certain category for aggregation training without using ground truth during training? I am eager to receive your suggestions and response.
Secondly,When I train my dataset, I set it to the same structure as mvtec.I have noticed the following situation: if I provide a more accurate mask, the prediction results and heat map obtained by the model during testing are more accurate, but when my mask is not so accurate, the results obtained are far inferior to the former. Why is this happening? In my understanding, the mask is only a tool for evaluating the quality of predictio
ns and should not have an impact on training. We look forward to your reply!
<img width="201" alt="901fc623c71e9eae664b154e15da15e" src="https://user-images.githubusercontent.com/98138999/230553846-cb8764eb-0883-45cd-81dc-2aba35092947.png">
<img width="185" alt="bb305a1dd6d46ffa2b22f2ca89fce13" src="https://user-images.githubusercontent.com/98138999/230553865-c2fa2b69-7793-4e9f-bfaf-b8c4958ef01d.png">
<img width="600" alt="9ac9dc25703ea87ee6a368ecf0c2418" src="https://user-images.githubusercontent.com/98138999/230553899-693f4bd6-125e-4c0a-8997-7ee0e030e557.png">
<img width="600" alt="f9d02f86d82022a4a1e79c5926e0332" src="https://user-images.githubusercontent.com/98138999/230553997-ff0b4e17-9853-4bb3-b4f7-059ef0fbafb3.png">

**Comment by chaoqinhuang** [comment]
1. During the testing phase, the results of each epoch of training are observed without making any changes to the parameters.
2. While visualization code is not provided in this project, you may use the code found in issue #14. It is worth noting that this code was borrowed from [RIAD](https://github.com/plutoyuxie/Reconstruction-by-inpainting-for-visual-anomaly-detection), and their visualization code uses a collection of masks to determine a threshold for localization across the entire dataset. If you only use one image for visualization, it is incorrect to choose the threshold based on only one ground-truth mask.
3. We have tried using a deeper backbone such as Res50, and it results in better performance.

**Comment by NAZABANMAAFK** [comment]
Thank you! I think I understand what you said. After using deeper networks for feature extraction, such as resnet50, the final feature dimensions are 1792(concat layer1~3), which do require more memory，have you used some dimensionality reduction methods? Furthermore，I only changed the parameters of conv1x1 in Encoder and Predictor to the corresponding dimensions(the dimention of layer3), and I would like to ask if I need to change anything else
![image](https://user-images.githubusercontent.com/98138999/231083231-997c0778-3796-49b6-820d-c30eaba1f819.png)

**Comment by chaoqinhuang** [comment]
We randomly selected a subset of the 1792 channels from Res50's features. The remaining number of channels was the same as that of Res18. Despite the loss of some channels due to random selection, we found that Res50 still outperformed Res18. Thus, we did not make any changes to the code parameters. We cannot confirm the performance that would result from using all dimensions of the features. May be better.
